bin/.upgrades/versions/v1/upgradeStorage.js

```diff
@@ -0,0 +1,389 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2018 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const
+  _ = require('lodash'),
+  getESConnector = require('../../connectors/es');
+
+const
+  INTERNAL_PREFIX = '%',
+  PUBLIC_PREFIX = '&',
+  NAME_SEPARATOR = '.';
+
+function getNewIndexName (index, collection) {
+  const prefix = index[0] === '%' ? '' : PUBLIC_PREFIX;
+
+  return `${prefix}${index}${NAME_SEPARATOR}${collection}`;
+}
+
+async function moveData (source, target, index, collection, newIndex) {
+  let page = await source.search({
+    index,
+    type: collection,
+    body: { sort: [ '_doc' ] },
```

Comment:
Why sorting on `_doc` and not on `_id` ?

Reply:
This is not really a sort, rather a performance trick when scrollign through an entire collection.

See: https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-body.html#request-body-search-scroll

(just before the "Keeping the search context alive" chapter)

Reply:
I didn't know this one!